enhancements/installer/agent-based-installation-in-hive.md

```diff
@@ -0,0 +1,647 @@
+ClusterDeployment's current version of "v1".
+
+#### Agent REST API Auth
+
+Agents will communicate with the assisted service using the non-kubernetes REST
+API that is currently in use by the SaaS. The backend assisted service will be
+listening on a socket that is routable from anywhere Agents are expected to
+run, so it needs to be secured from other potential actors on the network.
+
+When an Agent connects to report its existence, send logs, send inspection
+data, etc. it will need to authenticate.
+
+An authentication artifact will be embedded into the discovery ISO for use by
+the agent. As long as the user protects the contents of the discovery ISO
+itself, they can be confident that any client interacting with the backend
+service's REST API is the expected Agent running as part of the live discovery
+ISO. Details of the artifact are still being decided.
+
+#### Other REST API Access
+
+Where the assisted service has capabilities which are not (yet) expressed
+through kubernetes APIs, users may be tempted to access those capabilities
+directly via the assisted servuce REST API, thereby taking a dependency on the
+capability and its REST API. This would limit our ability to evolve those APIs
+in future.
+
+To mitigate this, the assisted service's REST API, except for the Agent API as
+described above, will limited to internal use only. It may be limited through
+authorization mechanisms or by blocking entire URL path space from external
+access.
+
+## Design Details
+
+### Open Questions
+
+#### Can we use sqlite?
+
+It would be advantageous to use sqlite for the database especially on
+stand-alone clusters, so that it is not necessary to deploy and manage
+an entire RDBMS, nor ship and support its container image. The [gorm library
+supports sqlite](https://gorm.io/docs/connecting_to_the_database.html#SQLite),
+but it is not clear if assisted service is compatible. In particular, the [use
+of FOR
+UPDATE](https://github.com/openshift/assisted-service/blob/e70af7dcf59763ee6c697fb409887f00ab5540f5/pkg/transaction/transaction.go#L8)
+might be problematic.
+
+For Hub clusters doing multi-cluster creation and management, there is an
+expectation that persistent storage availability and scale concerns will be a
+better fit for running a full RDBMS.
+
+A suggestion has been made that in situations where there is a single process
+running the assisted service, locking can happen in-memory instead of in the
+database. Further analysis is required.
+
+#### baremetal-operator watching multiple namespaces?
+
+When utilizing baremetal-operator on a Hub cluster to boot the discovery ISO on
+hosts, should we be creating those BareMetalHost resources in a separate
+namespace from those that are associated with the Hub cluster itself?
+
+What work is involved in having BMO watch additional namespaces?
+
+Upstream, the metal3 project is already running baremetal-operator watching
+multiple namespaces, so there should not be software changes required.
```

Comment:
The cluster-baremetal-operator does need to be updated to configure the baremetal-operator differently for this case.

Reply:
@hardys do we have something tracking that?

Reply:
I did add a sentence below this "cluster-baremetal-operator needs to be updated to configure baremetal-operator
differently for this use case."